benchmarks/src/main/java/org/elasticsearch/benchmark/index/codec/tsdb/ES87TSDBDocValuesBenchmark.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.benchmark.index.codec.tsdb;
+
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.AbstractDocValuesForUtilBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.DecodeConstantIntegerBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.DecodeDecreasingIntegerBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.DecodeFloatingPointBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.DecodeIncreasingIntegerBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.EncodeConstantIntegerBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.EncodeDecreasingIntegerBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.EncodeFloatingPointBenchmark;
+import org.elasticsearch.benchmark.index.codec.tsdb.internal.EncodeIncreasingIntegerBenchmark;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+@Fork(value = 1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 10)
+@BenchmarkMode(value = Mode.AverageTime)
+@OutputTimeUnit(value = TimeUnit.NANOSECONDS)
+@State(value = Scope.Benchmark)
+public class ES87TSDBDocValuesBenchmark {
+    @Param({ "4", "8", "12", "16", "24", "28", "32", "36", "40", "44", "48", "52", "56", "64" })
+    private int bitsPerValue;
+    @Param(
+        {
+            "decode constant integer",
+            "decode increasing integer",
+            "decode decreasing integer",
+            "decode floating point",
+            "encode constant integer",
+            "encode increasing integer",
+            "encode decreasing integer",
+            "encode floating point" }
+    )
+    private String mode;
```

Comment:
JMH is going to test every combination of bitsPerValue and mode, but a non-zero number of bits per value and constant integers don't really make sense together? Maybe collapse these two benchmark params into a single one?

Reply:
It is not clear to me what you means here:
* instead of using two parameters `bitsPerValue` and `mode` just use one (that encodes the combination) for all benchmarks?
*or just for the constant integer benchmark using the minimum number of bits necessary to represent the specific value of the constant integer?

I made two parameters and choose the verbose representation of the mode just to have a nicely formatted table of results at the end.